src/linalg/svd.rs

```diff
@@ -158,6 +163,16 @@ where
         let mut diagonal = bi_matrix.diagonal();
         let mut off_diagonal = bi_matrix.off_diagonal();
 
+        // **Check for NaN values in the diagonal elements**
+        // We check whether any singular value in the diagonal is NaN by comparing each value to itself.
+        // This works because NaN is the only value in Rust that is not equal to itself.
+        if diagonal.iter().any(|s| s != s) {
```

Comment:
did you choose to do it like this because the RealField and ComplexField traits don't expose an is_nan?